x-pack/plugins/upgrade_assistant/public/application/components/overview/backup_step/cloud_backup.tsx

```diff
@@ -22,16 +22,24 @@ import {
 } from '@elastic/eui';
 
 import { useAppContext } from '../../../app_context';
+import { ResponseError } from '../../../../../common/types';
+import { CLOUD_SNAPSHOT_REPOSITORY } from '../../../../../common/constants';
 import { uiMetricService, UIM_BACKUP_DATA_CLOUD_CLICK } from '../../../lib/ui_metric';
 
 interface Props {
   cloudSnapshotsUrl: string;
   setIsComplete: (isComplete: boolean) => void;
+  setForceOnPremStep: (forceOnPrem: boolean) => void;
 }
 
+const isMissingFoundSnapshotsRepo = (error: ResponseError) => {
+  return error.statusCode === 404 && error.message === `[${CLOUD_SNAPSHOT_REPOSITORY}] missing`;
```

Comment:
Nit: can we future-proof this by checking that the `message` includes the `CLOUD_SNAPSHOT_REPOSITORY` instead of comparing against the entire string? If it's a `404` then I think this would be sufficient to infer that the repository is missing, without need to also depend upon the specific language ES is using to communicate this.

Reply:
I think even safer would be to add a test for how missing repo error is returned to UA api integration tests [here](https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/apis/upgrade_assistant/cloud_backup_status.ts).

Reply:
Thanks both for having a look! I've updated this function to check if the message includes the repo name instead of matching against the entire string as suggested.